### PR TITLE
Adapting the pipeline for TTS usecases

### PR DESCRIPTION
**Pull request by Tomiinek**
This PR includes: 

- torch updates:
   - code and dependencies updated for torch and torchaudio >=2.8  
- refined VAD segmentation 
   - VAD is run on 80 ms chunks for better granularity
   - pre-bounce of speech onsets does not shift speech ends anymore
   - VAD is tweaked to be less sensitive to speech ends, avoiding mid-sentence cuts 
- bugfixes in speaker segmentation
   - run in multiple overlapping chunks for better granularity
   - fixed a bug that laed to cutting of segment ends to multiples of 340 ms  
- background music detection:
   - current voice separateion component has very high false-positive rate and even the sensitivity is not perfect
   - we trained a custom frame-wise background music detection model based on wav2vec2-BERT 
- dual-ASR verification:
   - added Whisper and Voxtral for getting multilingual transcripts
   - pipeline allows WER-based filtering and saves transcripts of the primary ASR

### CODE REVIEW

File: src/tidytunes/pipeline_components/music_detection.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    Returns:
+        A music detection model.
+    """
+    from tidytunes.models import MusicDetectionModel
+
+    detector = MusicDetectionModel()
+    detector.load_state_dict(
+        torch.load(
+            "/mnt/homes/tomiinek/various_scripts/checkpoints/best_model/model.pt"
```
Comment (janvainer) [inline src/tidytunes/pipeline_components/music_detection.py]: .

File: src/tidytunes/utils/etc.py [hunk 2]
```diff
@@ -38,12 +40,29 @@ def frame_labels_to_time_segments(
         frame_shift (float): Frame shift, used to convert frame indices to time stamps.
         filter_with (callable, optional): Filtering function of output segments.
                                           Defaults to no filtering.
+        segment_duration (float, optional): Duration of the original audio segment in
```
Comment (janvainer) [inline src/tidytunes/utils/etc.py:43]: Nice <3

**Review by janvainer — APPROVED** [review]

File: src/tidytunes/models/asr.py [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+        inputs = self.processor.apply_transcription_request(
+            language=language,
+            audio=[sample.cpu().numpy() for sample in audio_16khz],
+            model_id=self.model_name,
+            format=["WAV"] * B,
+        )
+        inputs = inputs.to(device, dtype=torch.bfloat16)
+
+        with torch.no_grad():
+            outputs = self.model.generate(**inputs, max_new_tokens=500)
```
Comment (srdecny) [inline src/tidytunes/models/asr.py:340]: >max_new_tokens=500

this is okay because the audio segments won't be long, but if you throw an hour long audio into it, it's going to get truncated

File: src/tidytunes/pipeline_components/asr_comparison.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    with torch.no_grad():
+        primary_results = primary_model(a, language, return_timestamps=False)
+        secondary_results = secondary_model(a, language, return_timestamps=False)
+
+    wer_values = []
+    for a, primary, secondary in zip(audio, primary_results, secondary_results):
+        wer = compute_wer(primary.text, secondary.text)
+        if a.origin is not None:
+            a.origin.transcript = primary.text
```
Comment (srdecny) [inline src/tidytunes/pipeline_components/asr_comparison.py:55]: You're using transcripts from the primary model which is Whisper, but isn't Voxtral slightly better?

Comment (srdecny) [reply src/tidytunes/pipeline_components/asr_comparison.py:55]: Also the text you're storing is not normalized

File: src/tidytunes/utils/asr.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+    try:
+        wer = jiwer.wer(
+            reference,
+            hypothesis,
+            reference_transform=transformation,
+            hypothesis_transform=transformation,
+        )
+        return wer
+    except Exception as e:
```
Comment (srdecny) [inline src/tidytunes/utils/asr.py]: uhh what is this? If this handles the case where one or more of the inputs is empty, I think it would be better to handle it more explicitly rather than catching every Exception

File: src/tidytunes/utils/audio.py [hunk 6]
```diff
@@ -112,6 +114,12 @@ def to_file(
         torchaudio.save(
             str(path), self.data.unsqueeze(0), self.sampling_rate, format="flac"
         )
+
+        if self.origin and self.origin.transcript:
```
Comment (srdecny) [inline src/tidytunes/utils/audio.py:116]: Maybe it would be useful to also store the ASR's name used for transcription in the path?

**Comment by Tomiinek** [comment]
I am hitting this issue https://github.com/pytorch/pytorch/issues/165861 
Which is fixed in torch 2.10, so I will also bump all the versions.

File: src/tidytunes/utils/audio.py (hunk 6 above)
Comment (Tomiinek) [reply src/tidytunes/utils/audio.py:116]: Yup, good idea.

File: src/tidytunes/models/asr.py (hunk 3 above)
Comment (Tomiinek) [reply src/tidytunes/models/asr.py:340]: Ok, the question is whether we want to use it for inputs longer then 30 seconds. The batching in general (and transformer's implementation) won't work in general for inputs long inputs.

File: src/tidytunes/pipeline_components/asr_comparison.py (hunk 4 above)
Comment (Tomiinek) [reply src/tidytunes/pipeline_components/asr_comparison.py:55]: 1. The choice is up to the user (config setting). I am debugging on Czech and Voxtral is muuuch worse actually. But yeah, not sure about English 👍 
2. The pipeline is for TTS, I dont think we want any normalization here.

File: src/tidytunes/utils/asr.py (hunk 5 above)
Comment (Tomiinek) [reply src/tidytunes/utils/asr.py]: Yeah you are absolutely right, I will punish Calude for this.
